### PR TITLE
Gh 2939 sc gas price

### LINKED ISSUE #2939: FSM optional gas_price for on-chain tx fee computation
Add the optional setting of gas price to be used for computing the fee to be paid. This is only applicable for on-chain transactions.

If both `fee` and `gas_price` are provided, use the bigger value.

### PR DESCRIPTION
**Pull request by velzevur**
Fixes #2939

### CODE REVIEW

File: apps/aehttp/test/aehttp_sc_SUITE.erl [hunk 1]
```diff
@@ -530,8 +597,10 @@ sc_ws_open_(Config, ChannelOpts0, MinBlocksToMine) ->
                  get_accounts_by_pubkey_sut(aeser_api_encoder:encode(account_pubkey, IPubkey)),
     {ok, 200, #{<<"balance">> := RStartAmt}} =
                  get_accounts_by_pubkey_sut(aeser_api_encoder:encode(account_pubkey, RPubkey)),
-    IAmt = 70000 * aec_test_utils:min_gas_price(),
-    RAmt = 40000 * aec_test_utils:min_gas_price(),
+    IAmt = maps:get(initiator_amount, ChannelOpts0,
```
Comment (velzevur) [inline apps/aehttp/test/aehttp_sc_SUITE.erl]: If provided, use the `initiator_amount` and `responder_amount` for opening amounts. This is really important if later on we expect the channel to have enough tokens to pay for the fat close mutual fee :)

File: apps/aehttp/test/aehttp_sc_SUITE.erl [hunk 2]
```diff
@@ -3552,8 +3627,14 @@ log_basename(Config) ->
...
                 filename:join([Protocol, "pinned_env"]);
-            changeable_fee ->
+            optional_fee ->
                 filename:join([Protocol, "changeable_fee"]);
+            optional_gas_price ->
+                filename:join([Protocol, "changeable_fee"]);
+            optional_fee_higher_than_gas_price ->
+                filename:join([Protocol, "changeable_fee_higher_than_gas_price"]);
+            optional_fee_lower_than_gas_price ->
+                filename:join([Protocol, "changeable_fee_higher_than_gas_price"]);
```
Comment (uwiger) [inline apps/aehttp/test/aehttp_sc_SUITE.erl]: Is this pathname correct?

File: docs/release-notes/next/GH-2939_sc_gas_price.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+* State Channel FSM: fixes fee computation according to transaction size.
+  Until now close mutual transactions were expecting a fee to be provided by
+  the initiating party or a default value of 20 000 gas had been used instead.
```
Comment (uwiger) [inline docs/release-notes/next/GH-2939_sc_gas_price.md]:
```suggestion
  the initiating party, or a default value of 20 000 gas would be used instead.
```

File: docs/release-notes/next/GH-2939_sc_gas_price.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+* State Channel FSM: adds support for an optional `gas_price` parameter for
+  all on-chain transactions, including the channel create one. This
+  `gas_price` is to be used in transaction fee computation: the fee is
+  `gas_price * transaction_gas_cost`. The `transaction_gas_cost` is a subject
```
Comment (uwiger) [inline docs/release-notes/next/GH-2939_sc_gas_price.md]:
```suggestion
  `gas_price * transaction_gas_cost`. The `transaction_gas_cost` is subject
```

File: apps/aechannel/src/aesc_fsm.erl [hunk 5]
```diff
@@ -1729,18 +1735,35 @@ new_onchain_tx(channel_slash_tx, Opts,
...
+    FeeSpecified = maps:is_key(fee, Opts),
+    GasPriceSpecified = maps:is_key(gas_price, Opts),
+    case FeeSpecified of
+        true when not GasPriceSpecified -> %% use preset fee
             apply(Mod, new, [Opts]);
+        true when GasPriceSpecified -> %% both fee and gas_price are specified
+            {ok, Tx1} = apply(Mod, new, [Opts]),
+            {ok, Tx2} = create_with_minimum_fee(Mod, Opts#{fee => 0}, % gas_price is already in Opts
+                                                CurrHeight),
+            %% use the tx with the bigger fee
```
Comment (hanssv) [inline apps/aechannel/src/aesc_fsm.erl]: Why the bigger? The user wants to pay extra?

Comment (velzevur) [reply apps/aechannel/src/aesc_fsm.erl]: Yes, in the sense of _I want to pay at least that amount or at least that amount of `gas_price`_

**Pull request merged**